### ISSUE TITLE
Router data hooks not always being called

### ISSUE THREAD

**Issue by mauro1855** [rdkcentral/Lightning-SDK#306]
Hi Lightning-SDK team, 

I found a strange behaviour related to the Router's 'Data providing' hooks, and after discussing with my colleagues we believe it to be a bug.

I am using the ```on()``` data hook to fetch data and inject it to my components. I started working on error handling and implemented the ```!``` route to handle errors. I was happy to see that if the promise passed to ```on()``` in a certain route is rejected, the error page is triggered.

Here is the potential bug: as soon as that happens and I try to enter the same broken route a second time, the ```on()``` of that route is no longer called and the router allows access to the route even though it should continue to redirect to the error handler.

Here is an example project which you can use to replicate the problem:
[com.mauro1855.RouterIssue.zip](https://github.com/rdkcentral/Lightning-SDK/files/8246328/com.mauro1855.RouterIssue.zip)

If you start the app, instructions are on screen.

I managed to prevent this bug from happening by explicitly setting at least one of these two options in platform settings:
```
"router": {
      "reuseInstance": false,
      "lazyCreate": true
}
```
... of course, once the app is deployed in the operator these settings are something app developers can't necessarily control.

Can you take a look?
SDK version 4.8.1

**Steps to reproduce**
1. Setup a route where the ```on()``` datahook is used
2. Return a rejected promise in the datahook.
3. Navigate to the route. Return back and go to the same route one second time. The bug happens on the second and subsequent times the route is accessed.

**Expected Behaviour**
The router should always execute ```on()``` (which in this case would always reject and always redirect to the ```!``` error handler)

**Actual Behaviour**
The router executes ```on()``` only once for the route, the second time that route is accessed the ```on()``` hook is not called and the router allows access to the route.

CC @Thomvl

**Comment by michielvandergeest** [comment]
Thanks for reporting this issue @mauro1855, we'll look into a it.

@sandeep-vedam can you take a look at the attached zipfile to reproduce the issue?

**Comment by michielvandergeest** [comment]
The fix in PR #312 has been released in version 4.8.3
